### PR TITLE
Group alerts by alert ref when same name

### PR DESCRIPTION
**Pull request by thc202**
Differentiate alerts with same name but different alert ref which are essentially different (even if they might incorrectly have the same name).

### CODE REVIEW

**Comment by psiinon** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [7b1a2d70\-0dd1\-4751\-8359\-9c2a8011f452](https://eu.ast.checkmarx.net/projects/641c4d0f-0dae-4786-973b-ead9338157fd/scans?id=7b1a2d70-0dd1-4751-8359-9c2a8011f452&branch=thc202%2Falert-tree-group-ref)
### Great job! No new security vulnerabilities introduced in this pull request


---
Use @Checkmarx to interact with Checkmarx PR Assistant. 
*Examples:*
`@Checkmarx how are you able to help me?`
`@Checkmarx rescan this PR`

File: zap/src/test/java/org/zaproxy/zap/extension/alert/AlertTreeModelUnitTest.java [hunk 1]
```diff
@@ -415,7 +497,22 @@ private static Alert newAlert(
...
+            int pluginId,
+            String alertRef,
+            int id,
+            String name,
+            String nodeName,
+            String uri,
+            int risk,
+            int confidence) {
         Alert alert = new Alert(pluginId, risk, confidence, name);
+        if (alertRef != null) {
```
Comment (kingthorin) [inline zap/src/test/java/org/zaproxy/zap/extension/alert/AlertTreeModelUnitTest.java:513]: Add a test where it is null?

Comment (thc202) [reply zap/src/test/java/org/zaproxy/zap/extension/alert/AlertTreeModelUnitTest.java:513]: It already is for existing alerts, only the ones being added have a custom ref.

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thank you!
